dds_web/api/errors.py

```diff
@@ -87,8 +87,19 @@ def __init__(
 class DatabaseError(exceptions.HTTPException):
     """Baseclass for database related issues."""
 
-    def __init__(self, message="The DDS encountered an Flask-SQLAlchemy issue.", project=None):
-        super().__init__(message)
+    def __init__(
+        self,
+        message="",
+        pass_message=False,
+        project=None,
+    ):
+
+        if pass_message and message != "":
+            super().__init__(message)
+        else:
+            super().__init__("The system encountered an error in the database.")
+            if message != "":
+                message = super.message
```

Comment:
Actually what I meant was this:
```
 def __init__(
        self,
        message="The system encountered an error in the database.",
        project=None,
    ):

        super().__init__(message)
```

Would that not be ok?

Reply:
This is essentially how it looked like before I started to change it. 

But that would result in the error message to be shown to the user or am I missing something? I assumed that, because the superclass is the HTTP.exception, whatever is passed as message to  `super().__init__(message) ` will be returned to the user. 

Wouldn't we need to take care of logging the database error within the `except` statements that raise the error in that case and not pass anything to the error itself?  

```
       except sqlalchemy.exc.SQLAlchemyError as err:
            raise DatabaseError(str(err))
```

Reply:
Yes in that case we would have not pass the info in when we raise. But ok, then how about:
```
    def __init__(
        self,
        message,
        pass_message=False,
        project=None,
    ):

        if pass_message and message:
            super().__init__(message)
        else:
            super().__init__("The system encountered an error in the database.")
```
I don't understand the 
```
            if message != "":
                message = super.message
```
part though.

Reply:
Yes, that was a mistake. I wanted to make sure that there is a non-empty message to log without typing out the default message twice. 

So I get it back from `super()` after setting it there. I also wasn't aware that in Python an empty string and not only _NULL_ evaluates to False - this somewhat simplifies the `if` statement.

Reply:
Ah okey. Then 
```
    def __init__(
        self,
        message,
        pass_message=False,
        project=None,
    ):

        if not (pass_message and message):
            message = "The system encountered an error in the database."
        
        super().__init__(message)
        general_logger.warning(message)
```

Reply:
Sorry for jumping in here but I think the point was to also be able to log the real error even though we send the vague message to the user?

Reply:
Oh ok! I didn't understand that that was what we were going for here. In that case I would probably log it first and then check if pass_message and then raise it with super().__init

Reply:
@MatthiasZepper